osu.Game.Rulesets.Catch/UI/CatchTouchInputMapper.cs

```diff
@@ -0,0 +1,302 @@
+            if (rightDashBox.Contains(inputPosition))
+                return TouchCatchAction.DashRight;
+            if (leftBox.Contains(inputPosition))
+                return TouchCatchAction.MoveLeft;
+            if (rightBox.Contains(inputPosition))
+                return TouchCatchAction.MoveRight;
+
+            return TouchCatchAction.None;
+        }
+
+        protected override void PopIn()
+        {
+            mainContent.FadeIn(500, Easing.OutQuint);
+        }
+
+        protected override void PopOut()
+        {
+            mainContent.FadeOut(300);
+        }
+
+        private class ArrowHitbox : CompositeDrawable, IKeyBindingHandler<CatchAction>
+        {
+            private readonly TouchCatchAction handledAction;
+
+            private readonly Box overlay;
+
+            private readonly Dictionary<object, TouchCatchAction> trackedActions;
+
+            private bool isHiglighted;
+
+            public ArrowHitbox(TouchCatchAction handledAction, ref Dictionary<object, TouchCatchAction> trackedActions, Color4 colour)
+            {
+                this.handledAction = handledAction;
+                this.trackedActions = trackedActions;
+
+                InternalChildren = new Drawable[]
+                {
+                    new Container
+                    {
+                        Width = 1,
+                        RelativeSizeAxes = Axes.Both,
+                        Children = new Drawable[]
+                        {
+                            new Box
+                            {
+                                Alpha = 0.8f,
+                                Colour = colour,
+                                Width = 1,
+                                RelativeSizeAxes = Axes.Both,
+                            },
+                            overlay = new Box
+                            {
+                                Alpha = 0,
+                                Colour = colour.Multiply(1.4f),
+                                Blending = BlendingParameters.Additive,
+                                Width = 1,
+                                RelativeSizeAxes = Axes.Both,
+                            }
+                        }
+                    }
+                };
+            }
+
+            public bool OnPressed(KeyBindingPressEvent<CatchAction> _)
+            {
+                if (trackedActions.ContainsValue(handledAction))
+                {
+                    isHiglighted = true;
+                    overlay.FadeTo(0.5f, 80, Easing.OutQuint);
+                }
+
+                return false;
+            }
+
+            public void OnReleased(KeyBindingReleaseEvent<CatchAction> _)
+            {
+                if (isHiglighted && !trackedActions.ContainsValue(handledAction))
+                {
+                    isHiglighted = false;
+                    overlay.FadeOut(1000, Easing.Out);
+                }
+            }
+        }
+
+        public enum TouchCatchAction
```

Comment:
I'm not a huge fan of this existing at all (replaced with arrays of keys throughout this file instead) but it does seem to work and I'm not really wanting to spend the time refactoring.

Just going to leave this as-is and see how things play out. Worst case we can refactor this class in the future.